diff --git a/setupbase.py b/setupbase.py
index 748b4dd6a..d07962991 100644
--- a/setupbase.py
+++ b/setupbase.py
@@ -223,7 +223,8 @@ def find_entry_points():
     return (
         [e % "" for e in ep]
         + [e % major_suffix for e in ep]
-        + [e % minor_suffix for e in ep]
+        # conda-forge: this is provided by _ipython_minor_entry_point
+        # + [e % minor_suffix for e in ep]
     )
 
 class install_lib_symlink(Command):
