diff --git a/setup.py b/setup.py
index 4939ca538..9cc5a7616 100644
--- a/setup.py
+++ b/setup.py
@@ -140,6 +140,12 @@
     'unsymlink': unsymlink,
 }
 setup_args["entry_points"] = {"console_scripts": find_entry_points()}
+# conda-forge
+setup_args["entry_points"]["pygments.lexers"] = [
+    "ipythonconsole = IPython.lib.lexers:IPythonConsoleLexer",
+    "ipython = IPython.lib.lexers:IPythonLexer",
+    "ipython3 = IPython.lib.lexers:IPython3Lexer",
+]
 
 #---------------------------------------------------------------------------
 # Do the actual setup now
